Libraries/LibWeb/Loader/ResourceLoader.cpp

```diff
@@ -272,18 +272,24 @@ void ResourceLoader::load(LoadRequest& request, GC::Root<SuccessCallback> succes
         // FIXME: Implement timing info for about requests.
         Requests::RequestTimingInfo fixme_implement_timing_info {};
 
+        auto serialized_path = url.serialize_path();
+
         // About version page
-        if (url.path_segment_at_index(0) == "version") {
+        if (serialized_path == "version") {
             success_callback->function()(MUST(load_about_version_page()).bytes(), fixme_implement_timing_info, response_headers, {}, {});
             return;
         }
 
         // Other about static HTML pages
-        auto resource = Core::Resource::load_from_uri(MUST(String::formatted("resource://ladybird/{}.html", url.path_segment_at_index(0))));
-        if (!resource.is_error()) {
-            auto data = resource.value()->data();
-            success_callback->function()(data, fixme_implement_timing_info, response_headers, {}, {});
-            return;
+        auto target_file = MUST(String::formatted("{}.html", serialized_path));
```

Comment:
Technically using serialized path is not correct as it could contain percent encoded characters. Though i image we won't put such files under here. See related https://github.com/LadybirdBrowser/ladybird/pull/3569

Reply:
Yeah I noticed that, although not percent decoding the path accidentally avoids a crash with non-utf-8 paths (such as `about:%ff`)

Reply:
It shouldn't be treated as a String, and should instead be a ByteString

Reply:
I've now changed it to percent decode the path and abort if the result isn't valid utf8

Reply:
Sorry, I hadn't seen your second comment yet, I've now changed it again to avoid converting to String (forgot that StringViews don't need to be utf8)